src/utils/constraints.ts

```
@@ -392,91 +530,188 @@ export const constrainToWalls = (
+      console.log(':::: isFlushMounted>>>>', isFlushMounted);
+      if (isFlushMounted) {
+        constrainedPosition.z = wallFaces.north;
+      } else {
+        constrainedPosition.z = wallFaces.north + halfDepth + wallBuffer;
+      }
+
+      console.log(' :::: constrainedPosition.z>>>>>', constrainedPosition.z, constrainedPosition);
+
+      // ✅ CRITICAL FIX: Only constrain X if object would actually extend beyond room bounds
+      const wouldExtendWest = position.x - halfWidth < interior.minX;
+      const wouldExtendEast = position.x + halfWidth > interior.maxX;
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)